tiledb/api/c_api/fragments_list/fragments_list_api_external.h

```diff
@@ -0,0 +1,111 @@
+/**
+ * @file tiledb/api/c_api/fragments_list/fragments_list_api_external.h
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2023 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file declares the fragments list section of the C API for TileDB.
+ */
+
+#ifndef TILEDB_CAPI_FRAGMENTS_LIST_EXTERNAL_H
+#define TILEDB_CAPI_FRAGMENTS_LIST_EXTERNAL_H
+
+#include "../api_external_common.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * C API carrier for a TileDB fragments list
+ */
+typedef struct tiledb_fragments_list_handle_t tiledb_fragments_list_t;
+
+/**
+ * Returns the uri of the fragment at the given index of a TileDB fragments
+ * list object.
```

Comment:
Documentation should state that the `uri` output has a life span subordinate to that of the fragment list. In particular, the user should not try to free the output string.